app/src/main/kotlin/com/simplemobiletools/calculator/activities/CryptoActivity.kt

```diff
@@ -0,0 +1,149 @@
+package com.simplemobiletools.calculator.activities
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.ConnectivityManager
+import android.os.Bundle
+import android.view.View
+import com.simplemobiletools.calculator.R
+import com.simplemobiletools.commons.extensions.copyToClipboard
+import com.simplemobiletools.commons.extensions.restartActivity
+import com.simplemobiletools.commons.extensions.value
+import com.simplemobiletools.commons.extensions.performHapticFeedback
+import com.simplemobiletools.commons.extensions.toast
+import com.simpletools.calculator.commons.activities.SimpleActivity
+import com.simpletools.calculator.commons.extensions.config
+import com.simpletools.calculator.commons.extensions.updateViewColors
+import com.simpletools.calculator.commons.helpers.Calculator
+import kotlinx.android.synthetic.main.activity_crypto.*
+import me.grantland.widget.AutofitHelper
+import android.widget.NumberPicker
+import com.simplemobiletools.calculator.helpers.CryptoCalculatorImpl
+import com.simpletools.calculator.commons.helpers.Formatter
+
+class CryptoActivity : SimpleActivity(), Calculator {
+    private var storedTextColor = 0
+    private var vibrateOnButtonPress = true
+    private var storedUseEnglish = false
+
+    private var cryptoFROM: String = ""
+    private var cryptoTO: String = ""
+
+    lateinit var calc: CryptoCalculatorImpl
+
+    @SuppressLint("MissingSuperCall")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_crypto)
+
+        calc = CryptoCalculatorImpl(this, applicationContext)
+        updateViewColors(crypto_holder, config.textColor)
+
+        getButtonIds().forEach {
+            it.setOnClickListener {
+                calc.numpadClicked(it.id); checkHaptic(it)
+            }
+        }
+
+        var numberPickerFROM: NumberPicker = numberPicker1
+        var numberPickerTO: NumberPicker = numberPicker2
+
+        // Values of each of the Number Pickers
+        val valuesFROM = arrayOf("Bitcoin", "Ethereum", "Ripple", "Litecoin", "USD")
+        val valuesTO = arrayOf("Ethereum", "Ripple", "Litecoin", "USD", "Bitcoin")
```

Comment:
I believe these can be set inside of arrays.xml in order to be more easily maintained [Documentation for reference](https://developer.android.com/samples/MediaRouter/res/values/arrays.html)

Reply:
I shall implement this refactoring. Thank you for the suggestion kind sir